bin/verify-testgrid-run.sh

```diff
@@ -0,0 +1,45 @@
+#!/bin/bash
+
+set -euo pipefail
+
+function log() {
+    echo "$1" 1>&2
+}
+
+function bail() {
+    log "$1"
+    exit 1
+}
+
+function main() {
+  local testgrid_run_json
+  testgrid_run_json=$(curl -s "https://api.testgrid.kurl.sh/api/v1/runs?searchRef=${STAGING_RELEASE}")
+
+  # check if run exist
+  if echo "${testgrid_run_json}" | jq -e '.total == 0' &>/dev/null; then
+    bail "Did not find testgrid run for release ${STAGING_RELEASE}"
+  fi
+
+  local run_id
+  local success_count
+  local failure_count
+  local total_runs
+  local skipped_test_count
+  run_id=$(echo "${testgrid_run_json}" | jq -r '.runs[].id')
```

Comment:
what if there are two? is that possible?

```
~ echo '{"runs": [{"id":"1"},{"id":"2"}]}' | jq -r '.runs[].id'
1
2
```

should you add some safeguard to check that length is 1?

```
~ echo '{"runs": [{"id":"1"},{"id":"2"}]}' | jq -r '.runs | length'
2
```

Reply:
It _should_ only find one run for a particular ref. Since we're using the short 7 digit GIT SHA (e.g. `f76becbb`) as part of the Ref this is unique enough to return one result. Do you foresee a scenario where there could be multiple ref IDs containing the same git SHA? As an example, our ref ID today looks like this: `STAGING-release-v2022.12.12-0-70fdddef-20230102121858`

Reply:
if there is a bug in the api and it returns an unexpected result. im suggesting safeguards here

Reply:
okay. Script will bail in the event testgrid's list runs API returns multiple runs.

Reply:
did you make this change? i dont see it

Reply:
@emosbaugh I implemented here: https://github.com/replicatedhq/kURL/pull/3867/files#diff-0c6dd97df28ca05e219e3f7f6cf203461efe649ca0f99d92ed4f57e1f1d6e6d6R27-R31